bin/build-dist

```diff
@@ -39,6 +37,9 @@ elif [[ -e Makefile.PL ]]; then
     # shellcheck disable=SC2012
     DIR=$(ls -td -- */ | head -n 1 | cut -d'/' -f1)
     mv "$DIR" "$BUILD_DIR"
+elif [ -e dist.ini  ]; then
+	dzil authordeps --missing | cpm install -
+    dzil build --no-tgz --in "$BUILD_DIR"
```

Comment:
For my use cases, I want `dist.ini` to be prioritized. I have distributions which include Makefile.PL etc, but if it's a `Dist::Zilla` distribution, it's `dzil build` that I explicitly want to test.

Reply:
hm there are multiple things:
- test `dzil build`
- test generated builder is on par with generated
- allow testing in docker < 5.20

Reply:
> allow testing in docker < 5.20

In this case I don't think `dzil` makes sense at all. I think you could build your dist via `dzil` in one job, upload a build artifact and the download the artifact in subsequent builds for testing.

`dzil` is an authoring tool and I expect authors to be working on a somewhat modern version of Perl. If it doesn't work on some old version of Perl, I think that's ok, as I will never be building and releasing on a really old version. For people who need to develop on a really old version, hopefully the repository contains the `Makefile.PL` etc, so that they can build without `dzil`.

Reply:
> test generated builder is on par with generated

Can you expand on this?

Reply:
re "expect authors" ... yes, authors works, but they may be developing for older versions as well. In my case I'm developing modules to easy (I hope) modernizing legacy code ... and here I met code bases using pretty old perls (though none using 5.8)

For my `dist.ini` copies `Build.PL` from build so my github actions can use it and run tests on perls < 5.20.

re "expand on this", I guess it's about those multiple things.

- test `dzil build` + "on par" (that one is missing indent)
there should be test which verifies that files copied by `dzil` from build are same as files in repository (maybe author forgot to run it)

- allow testing on docker < 5.20
Make possible to use `perldocker/perl-tester` with label < 5.20 to test repositories with `dist.ini` and some builder file.

and also as I mentioned, ideally so that it will be possible to create reusable workflow able to test almost anything

Reply:
To restate the problem, my workflow is "build once and test everywhere" and yours is "build and test everywhere". Is that an accurate statement?

Reply:
that is use case of using `auto-build-and-test-dist` with perl matrix